LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda

```diff
@@ -207,21 +208,23 @@ module ConstructAndSignVoteM where
 
       (RWST-bind
          (RWST-bind
-            RWST-get
-            (RWST-put "lSafetyData ∙= (safetyData1 & sdLastVote ?~ vote)"))   -- modifies the state returned by
-                                                                              -- RWST-get
-         (λ _ → RWST-return (inj₂ "Vote∙new vote mvsNew"))            -- The Unit returned by RWST-bind
-                                                                              -- via RWST-put is ignored
+            (RWST-gets id)                                                                -- Fetch the state.
+            (λ st → RWST-put (st & lSafetyData ∙~ ("safetyData1" & sdLastVote ?~ "vote")))-- Modify the state returned by RWST-get.
+         (λ _ → RWST-return (inj₂ "vote"))                                                -- The Unit returned by RWST-bind
+                                                                                          -- via RWST-put is ignored
+
+      Note that "vote" is: Vote.newWithSignature voteData author ledgerInfo
+                             (ValidatorSigner.sign validatorSigner ledgerInfo)
 
    Rewriting our goal with this yields (the annotations on the right
    show how we instantiate the rules in the next step):
 
      RWST-weakestPre
       (RWST-bind
          (RWST-bind                                                              = m
-            RWST-get
-            (RWST-put "lSafetyData ∙= (safetyData1 [ sdLastVote ?= vote ])"))
-         (λ _ → RWST-return (inj₂ "Vote∙new vote mvsNew"))               = f
+            (RWST-gets id)
+            (λ st → RWST-put (st & lSafetyData ∙~ ("safetyData1" & sdLastVote ?~ "vote")))
```

Comment:
I modified the shorthand somewhat to be more accurate (hopefully without too much impact on the readability). I think its important that, from reading this line, it is immediate that the expression is a function.